Add docs related to data package (#1072)
eventespresso/barista: packages/edtr-services/README.md

# EDTR services

This package contains the Event Editor (EDTR) specific services, mostly related to data services.

## Services provided

-   Contexts

    This package provides all the contexts needed for EDTR to work. It does so in [`ContextProvider.tsx`](./src/context/ContextProvider.tsx). There is also [`withEdtrContext`](./src/context/withEdtrContext.tsx) HOC which can be consumed to provide EDTR contexts to any component.

-   Data/Apollo Services

    All the data services required for EDTR are provided by this package. The reason for moving the data services fom EDTR domain to this package was that the services were consumed by other domains like REM.

## Data services

-   Cache rehydration

    The most important service that this package provides is the data rehydration of the Apollo cache. It reads the DOM data and updates/rehydrates the local cache to avoid network requests being sent for the initial data. It does so in [`useCacheRehydration`](./src/apollo/initialization/useCacheRehydration.ts).

-   Queries

>     It provides all the data queries required for EDTR. The queries and all the related utilities are places in `src/apollo/queries` directory, organized into entity types.

Reviewer: hmmm... maybe:

```suggestion
    It provides all the data queries required for EDTR. The queries and all the related utilities can be found in the `src/apollo/queries` directory, organized into entity types.
```

Author: Done, thanks

-   Mutations

>     It provides all the mutations required for EDTR. The mutations and all the related utilities are places in `src/apollo/mutations` directory, organized into entity types.

Reviewer: ```suggestion
    It provides all the mutations required for EDTR. The mutations and all the related utilities can be found in the `src/apollo/mutations` directory, organized into entity types.
```

Author: Done, thanks
